docs/seed-image-generation.md

```diff
@@ -0,0 +1,329 @@
+- OADP operator must be deployed.
+- Container storage must be setup as shared between stateroots, such as with a separate partition.
+- Required dnsmasq configuration to support updating cluster name, domain, and IP from the seed image as part of IBU.
+
+### Shared Container Storage
+
+Image Based Upgrade (IBU) requires that container storage (`/var/lib/containers`) is setup to be shared between stateroots. This can be done by creating a separate partition for container storage when the node is installed. Please see this [blog article](https://cloud.redhat.com/blog/a-guide-to-creating-a-separate-disk-partition-at-installation-time) for more information on how this is done using a `MachineConfig`.
+
+With ZTP GitOps, the `MachineConfig` can be added as `extra-manifests` under the site-config. For example:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 98-var-lib-containers-partitioned
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      disks:
+        - device: /dev/disk/by-id/wwn-0x62cea7f04d10350026c6f2ec315557a0
+          partitions:
+            - label: varlibcontainers
+              startMiB: 250000 # Leave room for rootfs
+              sizeMiB: 0 # Use available space
+      filesystems:
+        - device: /dev/disk/by-partlabel/varlibcontainers
+          format: xfs
+          mountOptions:
+            - defaults
+            - prjquota
+          path: /var/lib/containers
+          wipeFilesystem: true
+    systemd:
+      units:
+        - contents: |-
+            # Generated by Butane
+            [Unit]
+            Before=local-fs.target
+            Requires=systemd-fsck@dev-disk-by\x2dpartlabel-varlibcontainers.service
+            After=systemd-fsck@dev-disk-by\x2dpartlabel-varlibcontainers.service
+
+            [Mount]
+            Where=/var/lib/containers
+            What=/dev/disk/by-partlabel/varlibcontainers
+            Type=xfs
+            Options=defaults,prjquota
+
+            [Install]
+            RequiredBy=local-fs.target
+          enabled: true
+          name: var-lib-containers.mount
+```
+
+### Required dnsmasq Configuration
+
+Image Based Upgrade (IBU) requires that dnsmasq is configured such that it is able to use updated cluster name, domain, and IP address, as these will all be different for the target SNO compared to the seed SNO. The dnsmasq configuration is managed by a `MachineConfig` created during installation by the assisted-installer named `50-master-dnsmasq-configuration`. The changes required for IBU are not yet available in a GA ACM release, however, so a workaround is needed to override this `MachineConfig` until such time as we can define a minimum ACM version requirement.
```

Comment:
not sure we have to document it, no?

Reply:
This can be removed once we can define a minimum ACM version.